### PR TITLE
Issue #695 - RqFormBase: Get rid of the checkstyle suppression of ClassDataAbstractionCouplingCheck.

### PR DESCRIPTION
**Pull request by dalifreire**
Issue #695. Refactor the RqFormBase class to reduce the data abstraction coupling in order to get rid of the checkstyle suppression of ClassDataAbstractionCouplingCheck. This PR do the following changes:

- Adds a englishLowerCase method to the Utf8String class;
- Refactor the RqFormBase class and removes the ClassDataAbstractionCouplingCheck checkstyle suppression;

### CODE REVIEW

**Comment by dalifreire** [comment]
@davvd Please, find someone to review.

**Comment by davvd** [comment]
@dalifreire I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by davvd** [comment]
@mkordas this pull request is for you, please review

**Comment by mkordas** [comment]
@dalifreire I'll do a review here today

**Comment by mkordas** [comment]
@dalifreire I'm on it now, sorry for a bit of delay :)

File: src/main/java/org/takes/misc/Utf8String.java [hunk 1]
```diff
@@ -74,4 +75,12 @@ public Utf8String(final byte... bytes) {
     public String string() {
         return this.value;
     }
+
+    /**
+     * Returns the english lower case string representation.
+     * @return String value
+     */
+    public String englishLowerCase() {
+        return this.value.toLowerCase(Locale.ENGLISH);
```
Comment (mkordas) [inline src/main/java/org/takes/misc/Utf8String.java]: @dalifreire I have a concern here - this method does not depend on string being UTF8 or not, so it doesn't fit well this class I think

File: src/main/java/org/takes/rq/form/RqFormBase.java [hunk 2]
```diff
@@ -39,7 +39,6 @@
 import lombok.EqualsAndHashCode;
 import org.takes.HttpException;
 import org.takes.Request;
-import org.takes.misc.EnglishLowerCase;
```
Comment (mkordas) [inline src/main/java/org/takes/rq/form/RqFormBase.java]: @dalifreire so what are other usages of this class?

File: src/main/java/org/takes/rq/form/RqFormBase.java [hunk 3]
```diff
@@ -168,7 +163,7 @@ private static String decode(final CharSequence txt) {
                 );
             }
             final String key = decode(
-                new EnglishLowerCase(parts[0].trim()).string()
+                new Utf8String(parts[0].trim()).englishLowerCase()
```
Comment (mkordas) [inline src/main/java/org/takes/rq/form/RqFormBase.java]: @dalifreire looks like after your changes we'll have two ways of creating lowercase strings - either using dedicated class or Utf8String...

**Comment by mkordas** [comment]
@dalifreire see my comments above. BTW, maybe there's some less controversial dependency to remove?

**Comment by dalifreire** [comment]
@mkordas Firstly, thanks for review! 
So we need to remove at least one dependency... and I thought that was a good choice (why not?). Do you have another sugestion? The classes envolved are: 
- ByteArrayOutputStream,
- CopyOnWriteArrayList, 
- EnglishLowerCase, 
- HttpException, 
- RqPrint, 
- Sprintf, 
- Utf8String, 
- VerboseIterable

**Comment by mkordas** [comment]
@dalifreire the other option is just to split the class into two parts - have you investigated in this direction?

**Comment by mkordas** [comment]
@dalifreire what's your opinion here?

**Comment by dalifreire** [comment]
@mkordas Sorry about this late reply.. so, I think I found a better solution... please, could you review again?

**Comment by mkordas** [comment]
@dalifreire thanks, checking it

File: src/main/java/org/takes/rq/form/RqFormBase.java [hunk 4]
```diff
@@ -148,9 +143,9 @@ private static String decode(final CharSequence txt) {
      * @throws IOException If something fails reading or parsing body
      */
     private Map<String, List<String>> freshMap() throws IOException {
-        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        new RqPrint(this.req).printBody(baos);
-        final String body = new Utf8String(baos.toByteArray()).string();
+        final String body = new Utf8String(
+            new RqPrint(this.req).printBody()
+        ).string();
```
Comment (mkordas) [inline src/main/java/org/takes/rq/form/RqFormBase.java]: @dalifreire looks like this part is also already done in https://github.com/yegor256/takes/blob/master/src/main/java/org/takes/rq/RqPrint.java#L64

**Comment by mkordas** [comment]
@dalifreire see my comment above, I think we can make it even simpler

File: src/main/java/org/takes/rq/form/RqFormBase.java (hunk 4 above)
Comment (dalifreire) [inline src/main/java/org/takes/rq/form/RqFormBase.java]: @mkordas Good! you are right...

**Comment by dalifreire** [comment]
@mkordas Please, could you check again?

**Comment by mkordas** [comment]
@dalifreire looks good, thanks!

**Comment by mkordas** [comment]
@rultor merge

**Comment by mkordas** [comment]
@yegor256 ping

**Comment by mkordas** [comment]
@yegor256 we need your review here

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@ypshenychka please, review this task for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by ypshenychka** [comment]
@davvd Quality is good.

**Comment by davvd** [comment]
> @davvd Quality is good.

@ypshenychka thanks for the review and a good remark

**Comment by davvd** [comment]
@mkordas I added 10 mins to @ypshenychka (for QA review) in transaction 94805641... Much appreciated! **30 mins** was added to your account, payment ID `5798bacd2e745f177e6a6a6e`, time spent is 192 hours and 36 mins... review comments (c=15) added as a bonus... +30 to your rating, your total score is [+5223](http://www.netbout.com/b/36080?open=rating)

**Pull request merged**